tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -76,7 +85,35 @@ public String getMuleBin() {
 
   public void start(String... args) {
     checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(args);
-    osSpecificController.start(args);
+    try {
+      osSpecificController.start(args);
+    } catch (MuleControllerException mce) {
+      try {
+        printLog();
+      } catch (IOException ioe) {
+        mce.addSuppressed(ioe);
+      }
+
+      throw mce;
+    }
+  }
+
+  public void printLog() throws IOException {
+    final File muleLogFile = getLog();
+    if (muleLogFile.exists()) {
+      final StringBuilder muleLogBuilder = new StringBuilder();
+      muleLogBuilder.append(muleLogFile.getName() + ":" + lineSeparator());
+      muleLogBuilder.append("============" + lineSeparator());
+      try (BufferedReader br = new BufferedReader(new FileReader(muleLogFile))) {
+        String line;
+        while ((line = br.readLine()) != null) {
+          muleLogBuilder.append(" > " + line + lineSeparator());
+        }
+      }
+      LOGGER.info(muleLogBuilder.toString());
```

Comment:
To avoid the toString when info is disabled:
```suggestion
      LOGGER.info("{}", muleLogBuilder);
```